QRCodeReaderViewController/VerticalButton.m

```diff
@@ -0,0 +1,103 @@
+/*
+* VerticalButton
+*
+* Copyright 2014-present Yannick Loriot.
+* http://yannickloriot.com
+*
+* Permission is hereby granted, free of charge, to any person obtaining a copy
+* of this software and associated documentation files (the "Software"), to deal
+* in the Software without restriction, including without limitation the rights
+* to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+* copies of the Software, and to permit persons to whom the Software is
+* furnished to do so, subject to the following conditions:
+*
+* The above copyright notice and this permission notice shall be included in
+* all copies or substantial portions of the Software.
+*
+* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+* IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+* FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+* AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+* LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+* OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+* THE SOFTWARE.
+*
+*/
+
+#import "VerticalButton.h"
+
+@interface VerticalButton ()
+
+@property (nonatomic, strong) UILabel *textLabel;
+@property (nonatomic, strong) UIImageView *iconImageView;
+@property (nonatomic, strong) UITapGestureRecognizer *tapGesture;
+
+@end
+
+@implementation VerticalButton
+
+- (instancetype)initWithTitle:(NSString *)title image:(nullable UIImage *)image {
+  self = [super init];
+  if (self) {
+    self.title = title;
+    self.image = image;
+    self.translatesAutoresizingMaskIntoConstraints = false;
+    self.userInteractionEnabled = true;
+    
+    self.tapGesture = [[UITapGestureRecognizer alloc] init];
+    [self addGestureRecognizer: self.tapGesture];
+    
+    [self setupLayout];
+  }
+  return self;
+}
+
+- (void)addTarget:(id)target action:(SEL)action forControlEvents:(UIControlEvents)controlEvents {
+  [self.tapGesture addTarget:target action:action];
```

Comment:
- [ ] Since `VerticalButton` is a `UIControl` subclass, this should just work without overriding `-addTarget:action:forControlEvents:` and creating a gesture recogniser. You might need to set its subviews `userInteractionEnabled` to `false`.

Reply:
> * The button to open the photo library should be titled "Browse", not "Photos"

this controlled from init method, and updated in https://github.com/AlphaWallet/alpha-wallet-ios/pull/1823

Reply:
Are you referring to this line? https://github.com/AlphaWallet/alpha-wallet-ios/pull/1823/files#diff-179aa5097c92b30440628ea4ce183fc7R376

Reply:
yep